src/main/java/com/minecolonies/apiimp/initializer/InteractionValidatorInitializer.java

```diff
@@ -63,9 +63,9 @@ public static void init()
           citizen -> InventoryUtils.findFirstSlotInItemHandlerNotEmptyWith(citizen.getInventory(), ISCOOKABLE) != -1
                        && InventoryUtils.findFirstSlotInItemHandlerNotEmptyWith(citizen.getInventory(), stack -> CAN_EAT.test(stack) && (citizen.getWorkBuilding() == null || citizen.getWorkBuilding().canEat(stack))) == -1);
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(BETTER_FOOD),
-          citizen -> !citizen.isChild() && citizen.needsBetterFood());
+          citizen -> citizen.getSaturation() == 0 && !citizen.isChild() && citizen.needsBetterFood());
```

Comment:
you might want to do <= 0 since its a double and I'm not sure we're exactly setting it to zero explicitly

Reply:
We do, I've checked